[Chore] Resolve pre-existing CI debt surfaced by #403 (test fixtures, fmt, clippy)

**Difficulty:** Intermediate

## Context / Problem

PR #406 added `fmt` and `clippy` jobs to CI per #403, and reset the `Error` discriminants per #400 so the library now compiles. Running `cargo test --no-run --features testutils` and `cargo fmt --check` / `cargo clippy --all-targets --features testutils -- -D warnings` on the post-merge `main` surfaces three buckets of pre-existing debt that the new CI jobs will report as red until cleaned up. **None of this debt was introduced by #406** — it was masked on `main` because the duplicate-discriminant error (#400) blocked lib compilation before the test/lint pass could even reach it.

This issue tracks the cleanup so the new CI gates can become required checks.

## Bucket 1 — 23+ test fixture errors (`cargo test --no-run`)

Several test functions reference local bindings that were never declared in the fixture. The compiler emits `E0425: cannot find value` or `E0423: expected value, found module` for each. Locations:

- `src/test.rs:119-147` — missing `token_admin`, `contributor`, `creator`, and `token` is referenced as a value where it is a module.
- `src/tests/test_init.rs:33-59` — same missing bindings (`token_admin`, `contributor`, `creator`, `token`).
- `src/tests/test_admin.rs:25` — `error[E0382]: use of moved value: \`result\`` (the test consumes `result` twice in a row, e.g. by calling `.unwrap_err()` after the value has already moved).

Likely root cause: a refactor of the helper that builds test fixtures was applied to most callers but missed these. Compare against `src/tests/helpers.rs` and an adjacent test that *does* compile (e.g. `src/tests/test_contribute.rs`) to see the intended pattern.

## Bucket 2 — `cargo fmt -- --check` drift

Author of #406 reported approximately **148 lines** of rustfmt drift across `src/issues_test.rs`, `src/lib.rs`, and likely others. The fix is mechanical:

```bash
cargo fmt --all
```

Land the resulting diff in a single PR so the fmt job goes green. **Do not** mix fmt changes with logic changes — keeps the diff reviewable.

## Bucket 3 — `cargo clippy -- -D warnings`

Known pre-existing lints (non-exhaustive — run clippy to see the full list):

- `manual_div_ceil` in `src/lib.rs` (places that hand-roll `(a + b - 1) / b` instead of using `i128::div_ceil`).
- Several `dead_code` warnings for storage helpers that are no longer called (e.g. `get_block_contribution_count`, `set_block_contribution_count`, `get_category_campaigns`, `set_category_campaigns`, `increment_active_campaign_count` — confirmed by local `cargo check`). Cross-reference with #409 (DataKey cleanup tracking).

Fix in one PR (separate from Bucket 2 so reviewers can sanity-check each clippy diff individually).

## Suggested PR sequence

1. **Bucket 1 first** — restore the deleted bindings in `src/test.rs`, `src/tests/test_init.rs`, fix the `result` double-move in `src/tests/test_admin.rs`. Unblocks the `test` job.
2. **Bucket 2** — one `cargo fmt --all` PR. Unblocks the `fmt` job.
3. **Bucket 3** — clippy fixes, ideally per-lint or per-file. Unblocks the `clippy` job.

After all three land, mark `test`, `fmt`, and `clippy` as **required** status checks on `main` via branch protection. That closes the loop on #403.

## Expected Behavior

`cargo test --features testutils`, `cargo fmt --all -- --check`, and `cargo clippy --all-targets --features testutils -- -D warnings` all exit 0 on a fresh checkout of `main`. The new CI jobs from #406 are green.

## Acceptance Criteria

- [ ] `cargo test --features testutils` builds and runs all tests on `main`.
- [ ] `cargo fmt --all -- --check` exits 0.
- [ ] `cargo clippy --all-targets --features testutils -- -D warnings` exits 0.
- [ ] All three CI jobs (`test`, `fmt`, `clippy`) are marked as required checks in branch protection on `main`.

## Out of Scope

- Adding new tests — coordinate with #413 for test-tree consolidation.
- Removing dead `DataKey` variants — covered separately by the audit notes that fed #409. Just silencing the warning here is acceptable; structural deletion belongs in a later PR.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - `resume_campaign` now returns `ValidationFailed` when no pause is active, preventing spurious state writes and `campaign_resumed` events (#348).
 - `update_campaign` now emits both the updated title and description in `campaign_updated`, allowing full metadata indexing without extra reads (#349).
 
+### Infrastructure
+
+- Resolved pre-existing CI debt surfaced by the `fmt` and `clippy` gates added in #403: test fixture missing bindings restored in `src/test.rs` and `src/tests/test_init.rs`, `result` double-move fixed in `src/tests/test_admin.rs`, `cargo fmt --all` drift cleared across `src/issues_test.rs` and `src/lib.rs`, and clippy lints addressed (`manual_div_ceil` in `src/lib.rs`; `dead_code` suppressed on deferred storage helpers pending the DataKey audit in #409). All three CI jobs (`test`, `fmt`, `clippy`) now exit 0 on a clean checkout (#418).
+
 ### Refactored
 
 - Extracted `assert_admin(env, caller)` helper; used in `pause`, `unpause`, and `set_voting_params` to provide a single source of truth for admin authorization (#224).

